packages/webdriverio/tests/commands/browser/saveRecordingScreen.test.js

```diff
@@ -0,0 +1,50 @@
+import request from 'request'
+import { remote } from '../../../src'
+import * as utils from '../../../src/utils'
+
+describe('saveRecordingScreen', () => {
+    jest.mock('fs')
+    let browser
+    let getAbsoluteFilepathSpy
+    let assertDirectoryExistsSpy
+
+    beforeEach(async () => {
+        browser = await remote({
+            baseUrl: 'http://foobar.com',
+            capabilities: {
+                browserName: 'foobar',
+                mobileMode: true,
+                'appium-version': '1.11.1'
+            }
+        })
+
+        getAbsoluteFilepathSpy = jest.spyOn(utils, 'getAbsoluteFilepath')
+        assertDirectoryExistsSpy = jest.spyOn(utils, 'assertDirectoryExists')
+    })
+
+    afterEach(() => {
+        getAbsoluteFilepathSpy.mockClear()
+        assertDirectoryExistsSpy.mockClear()
+    })
+
+    it('should capture video', async () => {
+        const video = await browser.saveRecordingScreen('./packages/bar.mp4')
+
+        expect(getAbsoluteFilepathSpy).toHaveBeenCalledTimes(1)
+        expect(getAbsoluteFilepathSpy).toHaveBeenCalledWith('./packages/bar.mp4')
+        expect(assertDirectoryExistsSpy).toHaveBeenCalledTimes(1)
+        expect(assertDirectoryExistsSpy).toHaveBeenCalledWith(getAbsoluteFilepathSpy.mock.results[0].value)
+
+        expect(request.mock.calls[1][0].method).toBe('POST')
+        expect(request.mock.calls[1][0].uri.pathname).toBe('/wd/hub/session/foobar-123/appium/stop_recording_screen')
+        expect(video.toString()).toBe('some screenshot')
+    })
+
+    it('should fail if no filename provided', async () => {
+        const expectedError = new Error('saveRecordingScreen expects a filepath')
+
+        await expect(
+            browser.saveRecordingScreen()
+        ).rejects.toEqual(expectedError)
+    })
```

Comment:
Let's also have a check here that the existence of the filepath is being checked

Reply:
adding it at the moment to save screenshots and save recording. Nice catch, thanks!

Reply:
@christian-bromann done